packages/jsii-pacmak/lib/targets/python.ts

```diff
@@ -1580,4 +1590,22 @@ class PythonGenerator extends Generator {
 
         return undefined;
     }
+
+    private getAbstractBases(cls: spec.ClassType): spec.ClassType[] {
+        const abstractBases: spec.ClassType[] = [];
+
+        if (cls.base !== undefined) {
+            const base = this.findType(cls.base.fqn);
+
+            if (!spec.isClassType(base)) {
+                throw new Error("Class inheritence that isn't a class?");
```

Comment:
This isn't a super-useful error message.  Under what circumstances would we end up here?

Reply:
Theoretically never I think. We would end up here if a class inherited from something that wasn't a class, like a primitive type or an interface or something, which shouldn't be possible AFAIK. However the types technically allow it so it's possible that a bug or something causes the JSII gets into a state where it has a class inheriting from something that isn't a class.

This basically exists for two reasons:

* ``findType`` doesn't return a ``ClassType``, it returns a ``Type`` and using ``spec.IsClassType`` like this informs the compiler that base is a ``ClassType`` if it gets past this if statement.
* Emulates an ``assert`` statement to verify that some assumption of the world holds true, but that should never *not* be true.

Reply:
Okay, understood.